### PR TITLE
{core} Fix #17170: Replace distutils.version.LooseVersion with packaging.version.parse

### LINKED ISSUE #17170: distutils.version.LooseVersion should be replaced by packaging.version.Version
**Is your feature request related to a problem? Please describe.**

As pointed out by https://github.com/Azure/azure-cli-extensions/pull/2987#discussion_r572827296,  `distutils.version.LooseVersion` is legacy and conformant only to the superseded [PEP 386](https://www.python.org/dev/peps/pep-0386/) (by [PEP 440](https://www.python.org/dev/peps/pep-0440/)).

In edge build, the version is `2.20.0.post20210302071300`, which causes a failure at 

https://github.com/Azure/azure-cli/blob/b2c8e710170fe9141f81246edc4ffa5e84bdc3ef/src/azure-cli-core/azure/cli/core/util.py#L1179

```
'<' not supported between instances of 'int' and 'str'
```

because `post` is not a supported versioning convention of `distutils.version.LooseVersion`.

**Describe the solution you'd like**

We should migrate to [`packaging.version.Version`](https://packaging.pypa.io/en/latest/version.html#packaging.version.Version).

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Replace distutils.version.LooseVersion with packaging.version.Version
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by fengzhou-msft** [comment]
You need to add `packaging` as a dependency package to use it in CLI packages. `packaging` is actually vendored into [setuptools](https://github.com/pypa/setuptools/tree/a94ccbf404a79d56f9b171024dee361de9a948da/pkg_resources/_vendor/packaging) and you can use `from pkg_resources import parse_version`. We have existing [example](https://github.com/Azure/azure-cli/blob/1c176d39f0ba7da2f52b49f37f4253c87f88bb29/src/azure-cli-core/azure/cli/core/__init__.py#L625-L627).

**Comment by qwordy** [comment]
What are the differences among `distutils.version.LooseVersion`, `distutils.version.StrictVersion` and `packaging.version.Version`?

**Comment by PARADISSEEKR** [comment]
> What are the differences among `distutils.version.LooseVersion`, `distutils.version.StrictVersion` and `packaging.version.Version`?

`distutils.version.LooseVersion` use `re.compile(r'(\d+ | [a-z]+ | \.)', re.VERBOSE)` to match,`distutils.version.StrictVersion` use `re.compile(r'^(\d+) \. (\d+) (\. (\d+))? ([ab](\d+))?$',re.VERBOSE | re.ASCII)` to match,they are using  PEP 386.
`packaging.version.Version` use `re.compile(r"^\s*" + VERSION_PATTERN + r"\s*$", re.VERBOSE | re.IGNORECASE)` to match,it using PEP 440.
`packaging.version.parse` parse the given version string and return either a :class:`Version` object or a :class:`LegacyVersion` object depending on if the given version is a valid PEP 440 version or a legacy version.

File: src/azure-cli/setup.py [hunk 1]
```diff
@@ -136,6 +136,7 @@
     'javaproperties==0.5.1',
     'jsmin~=2.2.2',
     'jsondiff==1.2.0',
+    'packaging==20.9',
```
Comment (fengzhou-msft) [inline src/azure-cli/setup.py:139]: Use flexible compatible version range in `setup.py` if no impact of bugs in other versions as this file is used by `pip install`. The packages may be shared with other Python software.
```suggestion
    'packaging~=20.9',
```

**Review by fengzhou-msft — APPROVED** [review]

**Review by qwordy — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/util.py [hunk 2]
```diff
@@ -352,10 +352,10 @@ def _print(val=''):
         print(val, file=output)
 
     def _get_version_string(name, version_dict):
-        from distutils.version import LooseVersion  # pylint: disable=import-error,no-name-in-module
+        from packaging.version import parse  # pylint: disable=import-error,no-name-in-module
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:355]: [PEP 632](https://peps.python.org/pep-0632/#migration-advice) also mentions:

> `distutils.version` — use the `packaging` package

**Pull request merged**